### DISCUSSION TITLE
Using non-OSM sources for map tiles?

### DISCUSSION THREAD

**Discussion by BryanWall** [mapsforge/mapsforge#1309] [Q&A] [answered]
Is it possible to compile map tiles from another source into the Mapsforge binary map format? 

For example RapidAPI has a lot of different styles of map tiles. Some are listed on this OSM page of tile servers.

https://wiki.openstreetmap.org/wiki/Tile_servers

How can I create a local binary version from one of the tile servers that requires passing an API key?

**Comment by devemux86** [comment, marked as answer]
Mapsforge map files are created with [map-writer](https://github.com/mapsforge/mapsforge/blob/master/docs/Getting-Started-Map-Writer.md) using osm files as input.

For online bitmap tile sources, see the [DownloadCustomLayerViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/DownloadCustomLayerViewer.java).
